src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,139 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.controller.LinearQuadraticRegulator;
+import edu.wpi.first.math.estimator.KalmanFilter;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.system.LinearSystem;
+import edu.wpi.first.math.system.LinearSystemLoop;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.UnitModel;
+
+import static frc.robot.Constants.*;
+import static frc.robot.Constants.Shooter.*;
+import static frc.robot.Ports.Shooter.*;
+
+public class Shooter extends SubsystemBase {
+    private static Shooter INSTANCE;
+    private final UnitModel unitModel = new UnitModel(TICKS_PER_REVOLUTION);
+    private final WPI_TalonFX motor = new WPI_TalonFX(MOTOR);
+    private final LinearSystemLoop<N1, N1, N1> linearSystemLoop;
+    private double currentTime = 0;
+    private double lastTime = 0;
+
+    private Shooter() {
+        motor.setInverted(IS_INVERTED);
+        motor.setSensorPhase(IS_SENSOR_IN_PHASE);
```

Comment:
I found out today that there is no need to set the sensor phase of the motor for TalonFX, also, I think it's more descriptive `TalonFXInvertType.CounterClockwise` (also found out today)